js/fileupload.js

```diff
@@ -316,42 +316,60 @@ var insertImageInTinyMCE = function(editor, image) {
 if (typeof tinyMCE != 'undefined') {
     tinyMCE.PluginManager.add('glpi_upload_doc', function(editor) {
         editor.on('PastePreProcess', function(event) {
+            var content = event.content;
             //Check if data is an image
             if (isImageFromPaste(event.content)) {
                 stopEvent(event);
 
                 //extract base64 data
-                var base64 = extractSrcFromImgTag(event.content);
-
-                //transform to blob and insert into editor
-                if (base64.length) {
+                $('<div></div>').append(event.content).find('img').each(function() {
+                    var before = content.split(this.outerHTML)[0];
```

Comment:
it was hard flor me to read this part, maybe we can do some intermediate steps ? (like assign this to a img var, splitting, etc)